### PR TITLE
Refactor/add language for jobad and contact

### PR DESCRIPTION
**Pull request by cjpsycos**

### CODE REVIEW

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementMailEventListener.java [hunk 1]
```diff
@@ -5,31 +5,34 @@
...
     private static final String JOB_ADVERTISEMENT_REJECTED_TEMPLATE = "JobAdRejectedMail.html";
+    private static final String JOB_ADVERTISEMENT_CANCELLED_SUBJECT = "mail.jobAd.cancelled.subject";
     private static final String JOB_ADVERTISEMENT_CANCELLED_TEMPLATE = "JobAdCancelledMail.html";
 
-    // TODO To clarify which language should be used with the PO (JobAdvertisement, Contact, ...)
-    private static final Locale DEFAULT_LOCALE = Locale.GERMAN;
-
     // TODO To clarify which bcc address should be used with the PO (Taken from MailSenderProperties. Maybe move it in the MailSenderService)
-    private static final String BCC = "bcc@example.com";
+    private static final String BCC = "bcc@example.com";  // stellen-mediamatik@seco.admin.ch
```
Comment (michael-wirth) [inline application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementMailEventListener.java]: we need to define different BCC for each environment. the value shoudl be defined in the config server

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/dto/ContactDto.java [hunk 2]
```diff
@@ -3,31 +3,40 @@
...
+
+    @NotBlank
     private String phone;
-    @NotNull
+
+    @NotBlank
     private String email;
 
+    @NotBlank
+    private String languageIsoCode;
```
Comment (michael-wirth) [inline application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/dto/ContactDto.java]: Should we define **@Pattern** constraint as `[a-z]{2}` ?

Valid values are only "de", "fr", "it", "en". should we use an Enum instead?

File: web/src/main/resources/liquibase/changelog/20180211080000_added_entity_JobAdvertisement.xml [hunk 3]
```diff
@@ -100,6 +105,7 @@
             <column name="CONTACT_LAST_NAME" type="varchar(50)"/>
             <column name="CONTACT_PHONE" type="varchar(20)"/>
             <column name="CONTACT_EMAIL" type="varchar(50)"/>
+            <column name="CONTACT_LANGUAGE" type="varchar(5)"/>
```
Comment (michael-wirth) [inline web/src/main/resources/liquibase/changelog/20180211080000_added_entity_JobAdvertisement.xml]: 5 chars? do we need to support "de_CH"?

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementMailEventListener.java (hunk 1 above)
Comment (cjpsycos) [reply application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementMailEventListener.java]: agree it's in the todo above

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/dto/ContactDto.java (hunk 2 above)
Comment (cjpsycos) [reply application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/dto/ContactDto.java]: no enum. constraints can be done

File: web/src/main/resources/liquibase/changelog/20180211080000_added_entity_JobAdvertisement.xml (hunk 3 above)
Comment (cjpsycos) [reply web/src/main/resources/liquibase/changelog/20180211080000_added_entity_JobAdvertisement.xml]: didn't know how Locale is saved in the db. couldn't connect to it for checking. so we are save

**Pull request merged**